eureka2-write-server/src/main/java/com/netflix/eureka2/server/transport/tcp/replication/TcpReplicationHandler.java

```diff
@@ -40,35 +48,117 @@
  */
 public class TcpReplicationHandler implements ConnectionHandler<Object, Object> {
 
+    private static final Logger logger = LoggerFactory.getLogger(TcpReplicationHandler.class);
+
     private final WriteServerConfig config;
-    private final SelfInfoResolver SelfIdentityService;
+    private final SelfInfoResolver selfIdentityService;
     private final SourcedEurekaRegistry<InstanceInfo> registry;
-    private final EvictionQueue evictionQueue;
     private final WriteServerMetricFactory metricFactory;
 
+    private final BehaviorSubject<ReceiverReplicationChannel> channelSubject;
+
     @Inject
     public TcpReplicationHandler(WriteServerConfig config,
-                                 SelfInfoResolver SelfIdentityService,
+                                 SelfInfoResolver selfIdentityService,
                                  SourcedEurekaRegistry registry,
-                                 EvictionQueue evictionQueue,
                                  WriteServerMetricFactory metricFactory) {
         this.config = config;
-        this.SelfIdentityService = SelfIdentityService;
+        this.selfIdentityService = selfIdentityService;
         this.registry = registry;
-        this.evictionQueue = evictionQueue;
         this.metricFactory = metricFactory;
+        this.channelSubject = BehaviorSubject.create();
+
+        channelSubject
```

Comment:
TCP replication handler is shared by all replication connections. We might get three new replication connections first, with random failures later on. How this scan process resolves this?